src/target/linux.rs

```diff
@@ -238,12 +321,84 @@ impl<'a> ReadMemory<'a> {
         };
 
         if bytes_read == -1 {
-            // fixme: return a proper error type
-            return Err(Box::new(nix::Error::last()));
+            return Err(nix::Error::last());
         }
 
-        // fixme: check that it's an expected number of read bytes and account for partial reads
+        Ok(bytes_read)
+    }
+
+    /// Splits readOps to those that read from read protected memory and those that do not.
+    fn split_protected(
+        &self,
+        read_ops: &'a [ReadOp],
+    ) -> Result<(Vec<&'a ReadOp>, Vec<&'a ReadOp>), Box<dyn std::error::Error>> {
+        use std::cmp::Ordering;
+
+        let maps = self.target.memory_maps()?;
+
+        let mut protected_maps = maps
+            .iter()
+            .filter(|map| !map.is_readable)
+            .collect::<Vec<_>>();
+
+        // Not sure if required
+        protected_maps.sort_by(|a, b| a.address.0.cmp(&b.address.0));
```

Comment:
> I cannot find any resource that states so.

Yeah, it wasn't easy to find an authoritative source about this, but I think the Linux's [proc filesystem documentation](https://github.com/torvalds/linux/blob/master/Documentation/filesystems/proc.rst) can be considered one and the following statement can be interpreted as a guarantee that addresses always read in the growing order:

> Note: reading /proc/PID/maps or /proc/PID/smaps is inherently racy (consistent output can be achieved only in the single read call).
>
> This typically manifests when doing partial reads of these files while the memory map is being modified. Despite the races, we do provide the following guarantees:
>    1. The mapped addresses never go backwards, which implies no two regions will ever overlap.
>    2. If there is something at a given vaddr during the entirety of the life of the smaps/maps walk, there will be some output for it.

So I believe it should be safe to remove this line.